src/rest/connector/libs/apic/implementation.py

```diff
@@ -297,14 +297,21 @@ def post(self, dn, payload, expected_status_code=requests.codes.ok,
                                                     p=payload))
 
         # Send to the device
-        if isinstance(payload, dict):
-            response = self.session.post(full_url, json=payload, timeout=timeout,
-                                         verify=False)
-        else:
+        if xml_payload:
             response = self.session.post(full_url, data=payload, timeout=timeout,
                                          verify=False,
-                                         headers={'Content-type': 'application/json'})
-        output = response.json()
+                                         headers={'Content-type': 'application/xml'})
```

Comment:
what if `xml_payload` is True, but payload is `dict`? I think need to add error-handling for the case.

Reply:
If ```xml_payload = True``` and payload was ```dict``` a 400 error code would have been returned (Failed to parse post request}
However, you are right, a more targeted error works here.
Added the targeted error in this situation